packages/mongodb-chatbot-server/src/routes/responses/createResponse.ts

```diff
@@ -305,10 +357,60 @@ const convertToObjectId = (
 export const hasTooManyUserMessagesInConversation = (
   conversation: Conversation,
   maxUserMessagesInConversation: number
-) => {
+): boolean => {
   const numUserMessages = conversation.messages.reduce(
     (acc, message) => (message.role === "user" ? acc + 1 : acc),
     0
   );
   return numUserMessages >= maxUserMessagesInConversation;
 };
+
+const hasConversationUserIdChanged = (
+  conversation: Conversation,
+  userId?: string
+): boolean => {
+  return conversation.customData?.userId !== userId;
+};
+
+interface AddMessagesToConversationParams {
+  conversations: ConversationsService;
+  conversation: Conversation;
+  metadata?: Record<string, string>;
+  input: CreateResponseRequest["body"]["input"];
+  messages: Array<SomeMessage>;
+}
+
+const saveMessagesToConversation = async ({
+  conversations,
+  conversation,
+  metadata,
+  input,
+  messages,
+}: AddMessagesToConversationParams) => {
+  const finalMessages: Array<SomeMessage> = [];
+
+  if (typeof input === "string") {
+    finalMessages.push({
+      role: "user",
+      content: input,
+      metadata,
+    });
+  } else {
+    finalMessages.push(
+      // TODO: fix type here, but the gist is to filter out any function/tool calls
```

Comment:
dont filter out. keep those in DB.

any filtering handled by `GenerateResponse` func